client/CMakeLists.txt

```diff
@@ -11,8 +11,14 @@ endif (${WITH_SRV} STREQUAL ON)
 add_executable(mosquitto_pub pub_client.c ${shared_src})
 add_executable(mosquitto_sub sub_client.c sub_client_output.c ${shared_src})
 
-target_link_libraries(mosquitto_pub libmosquitto)
-target_link_libraries(mosquitto_sub libmosquitto)
+set(extra_libraries "")
+
+if (QNX)
+    set (extra_libraries socket)
+endif()
+
+target_link_libraries(mosquitto_pub libmosquitto ${extra_libraries})
+target_link_libraries(mosquitto_sub libmosquitto ${extra_libraries})
```

Comment:
according to cmake documentation, you can just call target_link_libraries multiple times.  Just leaving the existing lines, and doing 
```
if (QNX)
   target_link_libraries(mosquitto_pub socket)
   target_link_libraries(mosquitto_sub socket)
endif()